### PR TITLE
k3s: remove extra buildInputs

### PR DESCRIPTION
**Pull request by euank**
In the intial version of this package, I missed a few inputs that
weren't needed including a debug hook.

###### Motivation for this change

Review on  #84325 after it was merged pointed out I had an extra debug buildInput still.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by euank** [comment]
I can try to fix the aarch64 build on this pr or separately. k3s is supposed to compile and work on arm, so it should be pretty easy to fix.

**Comment by euank** [comment]
cc @flokli since these changes are in response to your review on #84325

File: pkgs/applications/networking/cluster/k3s/default.nix [hunk 1]
```diff
@@ -34,6 +34,7 @@ let
     url = "https://kubernetes-charts.storage.googleapis.com/traefik-${traefikChartVersion}.tgz";
     sha256 = "1aqpzgjlvqhil0g3angz94zd4xbl4iq0qmpjcy5aq1xv9qciwdi9";
   };
+  pkgMaintainers = [ maintainers.euank ];
```
Comment (flokli) [inline pkgs/applications/networking/cluster/k3s/default.nix:37]: please don't move this up here - it doesn't save us a single line, and people might want to step up maintaining individual packages from below.

**Review by flokli — CHANGES_REQUESTED** [review]
Also, please do the  other fixes in separate commits (can be part of the same PR)

File: pkgs/applications/networking/cluster/k3s/default.nix (hunk 1 above)
Comment (euank) [reply pkgs/applications/networking/cluster/k3s/default.nix]: Done. My thought process for that was that there really is only one package that matters here (all the intermediate ones are only part of creating the final build), so the set of maintainers for them should always be the same.

**Comment by euank** [comment]
Well, I'm unable to get `nix-build --arg crossSystem '{ config = "aarch64-linux"; }'` to actually work on my system so far, and I haven't figured out the failure yet.

It fails on building git well before it gets to k3s. I'll give getting it working on aarch64 a try as a separate PR when I have more time to figure out this issue.

**Comment by misuzu** [comment]
This broke `k3s` build: https://hydra.nixos.org/build/116153430

**Comment by euank** [comment]
Thanks for pointing that out @misuzu, appreciated! I'll look into it soon

**Comment by flokli** [comment]
@misuzu that hydra build was still from a version with `breakpointHook` installed.

I was able to compile `k3s` just fine. A (still scheduled) build on hydra can be found here: https://hydra.nixos.org/build/116387472

**Pull request merged**